### PR TITLE
Add new SMS command - LOOP LGS/CLOSED

### PR DESCRIPTION
**Pull request by pzadroga**
- Adding new SMS command in order to be able to switch between APS mode LGS/Closed via SMS.
- Adding also feedback about chosen mode when LOOP STATUS is queried.
- Refactor "magic strings" of APS mode and replace it with Enum.

### CODE REVIEW

**Comment by pzadroga** [comment]
Will work on code coverage tomorrow.

**Comment by MilosKozak** [comment]
are you sure that `ApsMode.OPEN.name == "open"` and not "OPEN"?

**Comment by pzadroga** [comment]
> are you sure that `ApsMode.OPEN.name == "open"` and not "OPEN"?

That's a good point, thanks. I forgot that considering the patch gets merged, shared preferences value will still be lower case...
Will improve that.

**Comment by MilosKozak** [comment]
generaly OK but I'd not use `.lowercase` everywhere. If potential source of future bugs.
Add migration to uppercase in MainApp and then use direct value

File: core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+enum class ApsMode() {
+    OPEN,
+    CLOSED,
+    LGS,
+    UNDEFINED;
+
+    companion object {
+
+        fun secureValueOf(stringValue: String): ApsMode {
```
Comment (MilosKozak) [inline core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt:11]: i'd prefer something like
`fun fromString(stringValue: String?) = values().firstOrNull { it.text == stringValue}  ?: UNDEFINED`

Comment (pzadroga) [reply core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt]: Done!

File: core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+enum class ApsMode() {
+    OPEN,
+    CLOSED,
+    LGS,
+    UNDEFINED;
+
+    companion object {
+
+        fun fromString(stringValue: String?) = values().firstOrNull { it.name == (stringValue?.uppercase() ?: UNDEFINED.name) } ?: UNDEFINED
+    }
```
Comment (MilosKozak) [inline core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt:12]: `(stringValue?.uppercase() ?: UNDEFINED.name)` is not necessary
simply `stringValue?.uppercase()`  is enough

Comment (pzadroga) [reply core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt:12]: Ooops, that was redundant. Simplified in next commit.

**Pull request merged**